src/movies.js

```diff
@@ -1,16 +1,249 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(arrayMovies) {
+    const directorArray = arrayMovies.map(function(movie) {
+        return movie.director;
+    })
+        return directorArray;
+    }
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(array1) {
+
+    if (array1.length === 0) {
+        return 0;
+      }
 
+    const spielbergDrama = array1.filter(function(movie) {
+      //checks to see if directore is Steven
+      if(movie.director ==="Steven Spielberg") {
+          //iterates over the genre key to see if drama
+          for(let i = 0; i < movie.genre.length; i ++) {
```

Comment:
A cleaner and shorter way to achieve the same would be using the `.includes()` method:
```js
function howManyMovies(someMovies) {
  return someMovies.filter( mov => mov.director === 'Steven Spielberg' && mov.genre.includes('Drama')).length;
}
```